packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -437,12 +464,17 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
             ? PendingBlobStatus.OfflinePendingUpload
             : PendingBlobStatus.OfflinePendingOp;
 
-        // Submit a BlobAttach op. It's important we submit this op now before returning the blob handle so the
-        // BlobAttach op is sequenced prior to any ops referencing the handle. Otherwise an invalid handle could be
-        // added to the document if the ops are not all successfully submitted upon reconnection.
-        // storageId may be undefined but since we are not connected we will have a chance to add it when reSubmit()
-        // is called
-        this.sendBlobAttachOp(entry.storageId, localId);
+        /**
+         * If we haven't already submitted a BlobAttach op for this localId and storageId combination, send it before
+         * returning the blob handle. This will make sure that the BlobAttach op is sequenced prior to any ops
+         * referencing the handle, otherwise, an invalid handle could be added to the document.
+         * storageId may be undefined but since we are not connected we will have a chance to add it when reSubmit()
+         * is called on reconnection.
+         */
+        if (entry.storageId === undefined || this.opsInFlight.get(entry.storageId)?.includes(localId) !== true) {
```

Comment:
Just check entry.status === `PendingBlobStatus.OfflinePendingUpload` vs `PendingBlobStatus.OfflinePendingOp`

Reply:
Changed the condition to `entry.status !== PendingBlobStatus.OnlinePendingOp` since that's the only state where there could have been an op sent for this entry.